spring-cloud-stream-binder-kinesis/src/test/java/org/springframework/cloud/stream/binder/kinesis/KinesisBinderTests.java

```diff
@@ -16,28 +16,58 @@
 
 package org.springframework.cloud.stream.binder.kinesis;
 
+import com.amazonaws.services.kinesis.model.DescribeStreamResult;
+import org.hamcrest.Matchers;
+import org.hamcrest.core.Is;
+import org.junit.Assert;
 import org.junit.ClassRule;
+import org.junit.Test;
 
+import org.springframework.cloud.stream.binder.Binder;
+import org.springframework.cloud.stream.binder.Binding;
 import org.springframework.cloud.stream.binder.ExtendedConsumerProperties;
 import org.springframework.cloud.stream.binder.ExtendedProducerProperties;
 import org.springframework.cloud.stream.binder.PartitionCapableBinderTests;
 import org.springframework.cloud.stream.binder.Spy;
 import org.springframework.cloud.stream.binder.kinesis.properties.KinesisBinderConfigurationProperties;
 import org.springframework.cloud.stream.binder.kinesis.properties.KinesisConsumerProperties;
 import org.springframework.cloud.stream.binder.kinesis.properties.KinesisProducerProperties;
+import org.springframework.integration.channel.DirectChannel;
 
 /**
  * @author Artem Bilan
+ * @author Jacob Severson
  *
  */
 public class KinesisBinderTests
-		extends PartitionCapableBinderTests<KinesisTestBinder, ExtendedConsumerProperties<KinesisConsumerProperties>, ExtendedProducerProperties<KinesisProducerProperties>> {
+		extends PartitionCapableBinderTests<KinesisTestBinder, ExtendedConsumerProperties<KinesisConsumerProperties>,
+		ExtendedProducerProperties<KinesisProducerProperties>> {
 
 	private final String CLASS_UNDER_TEST_NAME = KinesisBinderTests.class.getSimpleName();
 
 	@ClassRule
 	public static LocalKinesisResource localKinesisResource = new LocalKinesisResource();
 
+	@Test
+	@SuppressWarnings("unchecked")
+	public void testAutoCreateStreamForNonExistingStream() throws Exception {
+		Binder binder = getBinder();
+		DirectChannel output = new DirectChannel();
+		ExtendedConsumerProperties<KinesisConsumerProperties> consumerProperties = createConsumerProperties();
+		String testStreamName = "nonexisting" + System.currentTimeMillis();
+		Binding<?> binding = binder.bindConsumer(testStreamName, "test", output, consumerProperties);
+		binding.unbind();
+
+		DescribeStreamResult streamResult = localKinesisResource.getResource().describeStream(testStreamName);
+		String createdStreamName = streamResult.getStreamDescription().getStreamName();
+		int createdShards = streamResult.getStreamDescription().getShards().size();
+		String createdStreamStatus = streamResult.getStreamDescription().getStreamStatus();
+
+		Assert.assertThat(createdStreamName, Is.is(testStreamName));
+		Assert.assertThat(createdShards, Is.is(Matchers.greaterThan(0)));
+		Assert.assertThat(createdStreamStatus, Is.is("ACTIVE"));
```

Comment:
Static imports for JUnit, Hamcrest and Mockito are fine in tests:
```
<module name="AvoidStaticImport">
            <property name="excludes"
                      value="org.junit.Assert.*,org.mockito.Mockito.*,org.mockito.Matchers.*,org.hamcrest.Matchers.*,org.assertj.core.api.Assertions.*"/>
</module>
```

More over we there particular suppression:
```
<suppress files="[\\/]src[\\/]test[\\/]java[\\/]" checks="AvoidStaticImport" />
```

So, consider do not refactor this concern in the future, please.